### PR TITLE
Add Squirrel to the new ref panel

### PR DESCRIPTION
**Pull request by chrismwendt**
Resolves https://github.com/sourcegraph/sourcegraph/issues/33546

CleanShot 2022-04-07 at 20 24 20@2x

## Test plan

Ran locally and it worked

## App preview:
- [Link](https://sg-web-ref-panel-squirrel.onrender.com)

### CODE REVIEW

File: client/web/src/codeintel/useSearchBasedCodeIntel.ts [hunk 1]
```diff
@@ -261,3 +317,132 @@ async function executeSearchQuery(terms: string[]): Promise<SearchResult[]> {
...
+    hover?: string
+    def: Range
+    refs?: Range[]
+}
+
+interface Range {
+    row: number
+    column: number
+    length: number
+}
```
Comment (mrnugget) [inline client/web/src/codeintel/useSearchBasedCodeIntel.ts:414]: In other parts of `client/web/src/codeintel` we use `{ Range } from '@sourcegraph/extension-api-types'`. And I've had trouble keeping that apart from the JS builtin `Range`.

Can we prefix this `Range` here with `Local` or something?

File: client/web/src/codeintel/useSearchBasedCodeIntel.ts [hunk 2]
```diff
@@ -146,9 +176,35 @@ export async function searchBasedDefinitions({
...
+    const symbol = await findSymbol({ repository: repo, commit, path, row: position.line, column: position.character })
+    if (symbol?.def) {
+        return [
+            {
+                repo,
+                file: path,
+                content: fileContent,
+                commitID: commit,
+                range: rangeToExtensionRange(symbol.def),
+                url: toPrettyBlobURL({
```
Comment (mrnugget) [inline client/web/src/codeintel/useSearchBasedCodeIntel.ts:192]: TIL about `toPrettyBlobURL`. I think I can use that in other parts... 🤔

**Review by mrnugget — APPROVED** [review]

**Comment by mrnugget** [comment]
@chrismwendt regarding the test plan in the description ("Existing tests"): since there are no tests for these hooks, I assume you also tested this locally and confirmed it works?

File: client/web/src/codeintel/useSearchBasedCodeIntel.ts (hunk 1 above)
Comment (chrismwendt) [reply client/web/src/codeintel/useSearchBasedCodeIntel.ts:414]: Yep, renamed to `LocalRange`

**Comment by chrismwendt** [comment]
Yep, I ran it locally ✅

**Pull request merged**